juriscraper/opinions/united_states/state/ri.py

```diff
@@ -0,0 +1,85 @@
+            f'<Object Type="String">{self.search_key}</Object>'
+            '</Parameter><Parameter Type="Array"><Object ObjectPathId="597" /></Parameter><Parameter Type="Boolean">true</Parameter></Parameters></Method></TryScope><CatchScope Id="654" /></ExceptionHandlingScope></Actions><ObjectPaths><Constructor Id="597" TypeId="{80173281-fffd-47b6-9a49-312e06ff8428}" /><Property Id="602" ParentId="597" Name="SortList" /><Property Id="605" ParentId="597" Name="RefinementFilters" /><Property Id="614" ParentId="597" Name="Properties" /><Property Id="619" ParentId="597" Name="SelectProperties" /><Property Id="626" ParentId="597" Name="HitHighlightedProperties" /><Constructor Id="646" TypeId="{8d2ac302-db2f-46fe-9015-872b35f15098}" /></ObjectPaths></Request>'
+        )
+
+        self.json_url = (
+            "https://www.courts.ri.gov/_vti_bin/client.svc/ProcessQuery"
+        )
+
+    def _download(self, request_dict={}):
+        if self.test_mode_enabled():
+            self.json = super()._download(request_dict)
+            return
+        return super()._download(request_dict)
+
+    def fetch_json(self) -> None:
+        """Fetch JSON data from the site.
+
+        :return: None
+        """
+        if self.test_mode_enabled():
+            return
+        self.parameters = self.data_xml
+        content = html.tostring(self.html)
+        request_digest = (
+            str(content).split("formDigestElement.value")[3].split("'")[1]
+        )
+        self.request["headers"]["X-RequestDigest"] = request_digest
+        self.request["headers"]["X-Requested-With"] = "XMLHttpRequest"
+        self._request_url_post(self.json_url)
+
+    def _process_html(self) -> None:
+        """Extract content from JSON response
+
+        :return: None
+        """
+        self.fetch_json()
```

Comment:
Getting an error when actually running the script

```
Traceback (most recent call last):
  File "/home/grossir/venvs/juriscraper38/lib/python3.8/site-packages/requests/models.py", line 971, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python3.8/json/__init__.py", line 337, in loads
    raise JSONDecodeError("Unexpected UTF-8 BOM (decode using utf-8-sig)",
json.decoder.JSONDecodeError: Unexpected UTF-8 BOM (decode using utf-8-sig): line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "sample_caller.py", line 263, in main
    site.parse()
  File "/home/grossir/Desktop/Proyectos/freelaw_repos/juriscraper/juriscraper/AbstractSite.py", line 140, in parse
    self._process_html()
  File "/home/grossir/Desktop/Proyectos/freelaw_repos/juriscraper/juriscraper/opinions/united_states/state/ri.py", line 73, in _process_html
    rows = self.request["response"].json()[-3][self.search_key][
  File "/home/grossir/venvs/juriscraper38/lib/python3.8/site-packages/requests/models.py", line 975, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Unexpected UTF-8 BOM (decode using utf-8-sig): line 1 column 1 (char 0)
```

The content of the json response
```
'\ufeff[\r{\r"SchemaVersion":"15.0.0.0","LibraryVersion":"16.0.10388.20004","ErrorInfo":{\r"ErrorMessage":"Unexpected end of file while parsing Name has occurred. Line 1, position 9896.","ErrorValue":null,"TraceCorrelationId":"bb2638a1-1ad6-a0d3-9c5d-3866ba4ed61b","ErrorStackTrace":"   at System.Xml.XmlTextReaderImpl.Throw(Exception e)\\r\\n   at System.Xml.XmlTextReaderImpl.ParseQName(Boolean isQName, Int32 startOffset, Int32& colonPos)\\r\\n   at System.Xml.XmlTextReaderImpl.ThrowTagMismatch(NodeData startTag)\\r\\n   at System.Xml.XmlTextReaderImpl.ParseEndElement()\\r\\n   at System.Xml.XmlTextReaderImpl.ParseElementContent()\\r\\n   at System.Xml.XsdValidatingReader.Read()\\r\\n   at Microsoft.SharePoint.Client.ClientXmlReader.Read()\\r\\n   at System.Xml.XmlLoader.LoadNode(Boolean skipOverWhitespace)\\r\\n   at System.Xml.XmlLoader.LoadDocSequence(XmlDocument parentDoc)\\r\\n   at System.Xml.XmlDocument.Load(XmlReader reader)\\r\\n   at Microsoft.SharePoint.Client.ClientMethodsProcessor.LoadRequestXml(Stream batchRequest, ClientServiceHost host)\\r\\n   at Microsoft.SharePoint.Client.ClientMethodsProcessor..ctor(Stream batchRequest, ClientServiceHost host, String boundary, IList`1 pendingDisposableQueue)\\r\\n   at Microsoft.SharePoint.Client.ClientRequestServiceImpl.ProcessQuery(Stream inputStream, IList`1 pendingDisposableContainer)","ErrorCode":-1,"ErrorTypeName":"System.Xml.XmlException"\r},"TraceCorrelationId":"bb2638a1-1ad6-a0d3-9c5d-3866ba4ed61b"\r}\r]'
```

Reply:
I am still getting this error for `ri`. I have tried with different IPs. Not sure what the problem may be. The request goes through when using my browser, but the XML is slightly different

Reply:
perhaps you can slack me the xml you have and I can see if the reverse is true and can be tested.